spyder/plugins/completion/languageserver/client.py

```diff
@@ -393,7 +402,7 @@ def send(self, method, params, kind):
                 return int(_id)
             except zmq.error.Again:
                 if time.time() > timeout_time:
-                    self.sig_lsp_down.emit(self.language)
+                    self.sig_went_down.emit(self.language)
```

Comment:
Should we actually kill the process at this stage?

Reply:
I also refactored this.

Reply:
That's handled by the restart mechanism.